packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -49,19 +62,35 @@ const VirtualizedComboBoxMenu = (props: React.ComponentProps<'div'>) => {
     );
   }, [focusedIndex, menuRef]);
 
-  const { outerProps, innerProps, visibleChildren } = useVirtualization({
-    // 'Fool' VirtualScroll by passing length 1
-    // whenever there is no elements, to show empty state message
-    itemsLength: filteredOptions.length || 1,
-    itemRenderer: virtualItemRenderer,
-    scrollToIndex: focusedVisibleIndex,
-  });
+  useLayoutEffect(() => {
+    if (focusedVisibleIndex) {
+      virtualizer.scrollToIndex(focusedVisibleIndex, { align: 'start' });
+    }
+  }, [focusedVisibleIndex, virtualizer]);
 
   return (
-    <Box as='div' {...outerProps} {...rest}>
-      <div {...innerProps} ref={innerProps.ref}>
-        {visibleChildren}
-      </div>
+    <Box
+      as='div'
+      style={{
+        minBlockSize: virtualizer.getTotalSize(),
+        minInlineSize: '100%',
+      }}
+      {...rest}
+    >
+      {virtualizer.getVirtualItems().map((virtualItem) => {
+        return React.cloneElement(virtualItemRenderer(virtualItem.index), {
+          key: virtualItem.key,
+          'data-index': virtualItem.index,
```

Comment:
Is this index different from the already-present `data-iui-index`?

Reply:
Yeah, that `data-index` value is what tanstack virtual uses for its default `measureElement` function [(docs)](https://tanstack.com/virtual/latest/docs/api/virtualizer#measureelement-1). Since we are now using a custom function for `measureElement`, though, we may be able to use our `data-iui-index` instead. I'll test and see if this works the way I think it should.